code/clue.js

```diff
@@ -1,59 +1,304 @@
+  document.getElementById("killerCard").style.background = mystery.killer.color;
+  document.getElementById("killerName").innerHTML = `${mystery.killer.name}`;
+  document.getElementById("killerImage").src = mystery.killer.image;
+  document.getElementById("killerDescription").innerHTML = `${mystery.killer.description}`;
+  document.getElementById("faveWeapon").innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon.name}`;
+  
+  //Shows the details of the killer
+  document.getElementById("cardDetails1").style.display = "flex"; 
+};
+
+//----- Weapon card -----//
+const pickWeaponLoader = () => {
+  document.getElementById("loader2").classList.add("animated-loader");
+  setTimeout(showPickWeapon, 2000);
+};
+
+document.getElementById("weaponCard").addEventListener("click", pickWeaponLoader); 
+
+const showPickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+  //find which weapon is the favourite for the killer card that has been chosen and make that weapon more likely to be put into mystery.weapon
+
+  document.getElementById("loader2").style.display = "none"
+  document.getElementById('weaponCard').removeEventListener('click', pickWeaponLoader);
+
+  document.getElementById("weaponCard").style.background = mystery.weapon.color;
+  document.getElementById("weaponName").innerHTML = `${mystery.weapon.name}`;
+  document.getElementById("weaponImage").src = mystery.weapon.image;
+  document.getElementById("weaponLastSeen").innerHTML = `Last seen in the ${mystery.weapon.lastseen}`;
+  document.getElementById("weaponStrength").innerHTML = `Strength: ${mystery.weapon.strength}`;
+
+  document.getElementById("cardDetails2").style.display = "flex";
+};
+
+//----- Room card -----//
+const pickRoomLoader = () => {
+  document.getElementById("loader3").classList.add("animated-loader");
+  setTimeout(showPickRoom, 2000);
+};
+
+document.getElementById("roomCard").addEventListener("click", pickRoomLoader); 
+
+const showPickRoom = () => {
+  mystery.room = randomSelector(rooms);
+  document.getElementById("loader3").style.display = "none"
+  document.getElementById('roomCard').removeEventListener('click', pickRoomLoader);
+
+  document.getElementById("roomCard").style.background = mystery.room.color;
+  document.getElementById("roomName").innerHTML = `${mystery.room.name}`;
+  document.getElementById("roomImage").src = mystery.room.image;
+  document.getElementById("roomLastSeen").innerHTML = `Last seen there: ${mystery.room.lastseen}`;
+  document.getElementById("roomfave").innerHTML = `Favourite room: ${mystery.room.favespot}`;
+
+  document.getElementById("cardDetails3").style.display = "flex";
+};
+
+//----- Function for on button click reveal mystery or get a message asking you to choose each card to then beable to reveal the mystery-----//
+function revealMystery() {
+  if (mystery.killer.hasOwnProperty("name") === true) {
```

Comment:
This is fine enough :

```
if (mystery.killer.name)
```

Reply:
@EmOlsz I changed the mystery object properties to null. So when I went to change what you suggested in the if statement I got an error message "Cannot read property 'name' of null at revealMystery". So changed it to "mystery.killer != null" instead so it would work inline with the mystery object properties being set to null. And if the user clicks on the first card it won't be null anymore, otherwise if they try to revel the mystery before clicking on the cards then they will get the message from the else statement. I hope it's ok to write this instead of what you suggested?